docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+
+- Sender must be a blacklister.
+
+**removeFromWhitelist** address
+
+- Remove address from whitelist.
+
+- Sender must be a blacklister.
+
+**removeFromWhitelistBatch** (list address)
+
+- Remove provided addresses from whitelist.
+
+- If some of specified addresses are not in whitelist, then
+  it fails returning list of those addresses.
+
+- Sender must be a blacklister.
+
+**checkWhitelisted** (void address bool)
+
+- Return a boolean value describing whether the
+  given address is present in whitelist.
+
+Role reassigning functions
+==========================
+
+**Owner**
+---------
+
+**transferOnership**
+
+Types
+```
+transferOwnership =
+  ( address :owner
+  , address :receiver
+  )
+```
+
+Parameter (in Michelson):
+```
+(pair %transferOwnership
+  (address %owner)
+  (address %receiver)
+)
+```
+
+- Set token owner to a new address.
+
+- Sender must be current token owner.
+
+- The current owner retains his priveleges up until
+  `acceptOwnership` is called.
+
+- Can be called multiple times, each call replaces pending
+  owner with the new one. Note, that if proposed
+  owner is the same as the current one, then the call
+  is simply invalidated.
+
+**getOwner**
+
+Types
+```
+getOwner = (contract address :callback)
+```
+
+Parameter (in Michelson):
+```
+(pair %getOwner
+  (contract %callback address)
+)
+```
+
+- Returns current token owner address.
+
+**acceptOwnership**
+
+Types
+```
+acceptOwnership = unit
+```
+
+Parameter (in Michelson):
+```
+(unit %acceptOwnership)
+```
+
+- Accept ownership privileges.
+
+- Sender must be a pending owner.
+
+**Admin**
+---------
+
+**changeTokenAdmin** address
+
+Types
+```
+changeTokenAdmin =
+  ( address :owner
+  , address :receiver
+  )
+```
+
+Parameter (in Michelson):
+```
+(pair %changeTokenAdmin
+  (address %owner)
+  (address %receiver)
+)
+```
+
+- Set token administrator to a new address.
+
+- Sender must be current token admin.
+
+- The current administrator retains his priveleges up until
+  `acceptAdminRole` is called.
+
+- Can be called multiple times, each call replaces pending
+  administrator with the new one. Note, that if proposed
+  administrator is the same as the current one, then the call
+  is simply invalidated.
+
+**getAdmin**
+
+Types
+```
+getAdmin = (contract address :callback)
+```
+
+Parameter (in Michelson):
+```
+(pair %getAdmin
+  (contract %callback address)
+)
+```
+
+- Returns current token administrator address.
+
+**acceptAdminRole**
+
+Types
+```
+acceptAdminRole = unit
+```
+
+Parameter (in Michelson):
+```
+(unit %acceptAdminRole)
+```
+
+- Accept administrator privileges.
+
+- Sender must be a pending administrator.
+
+**Master Minter**
+---------------
+
+**changeMasterMinter**
+
+Types
+```
+changeMasterMinter =
+  ( address :owner
+  , address :receiver
+  )
+```
+
+Parameter (in Michelson):
+```
+(pair %changeMasterMinter
+  (address %owner)
+  (address %receiver)
+)
+```
+
+- Set master minter to a new address.
```

Comment:
But earlier in the spec you assumed that there would be several master minters. Have you abandoned this idea?

Reply:
Nope, I'll clarify this